server/heroku/logs.go

```diff
@@ -24,9 +25,10 @@ func (h *PostLogs) ServeHTTPContext(ctx context.Context, w http.ResponseWriter,
 	}
 
 	var form PostLogsForm
-
 	if err := Decode(r, &form); err != nil {
-		return err
+		if err.Error() != "EOF" {
+			return fmt.Errorf("error decoding request: %v", err)
```

Comment:
It'd be nice if we did this globally across the handlers, but then you can't check for io.EOF. We could split out the Decode method so that we have a lower level DecodeRequest method that can ignore io.EOF.

e.g.:

```go
func DecodeRequest(r *http.Request, v interface{}, ignoreEOF) error {
        if err := json.NewDecoder(r.Body).Decode(v); err != nil {
                if err == io.EOF && ignoreEOF {
                        return nil
                }

                return fmt.Errorf("error decoding request body: %v", err)
        }

        return nil
}

func Decode(r *http.Request, v interface{}) error {
        return DecodeRequest(r, b, false)
}
```

Then, in this endpoint:

```go
ignoreEOF := true
if err := DecodeRequest(r, &form, ignoreEOF); err != nil {
        return err
}
```